### PR TITLE
fix: deprecate submitHashrate

### PR DESCRIPTION
**Pull request by BelfordZ**
re: https://github.com/multi-geth/multi-geth/issues/115

### CODE REVIEW

File: openrpc.json [hunk 1]
```diff
@@ -965,6 +965,7 @@
       },
       {
         "name": "eth_submitHashrate",
+        "deprecated": true,
         "summary": "Returns an array of all logs matching a given filter object.",
```
Comment (shanejonas) [inline openrpc.json]:
```suggestion
        "summary": "Used for submitting mining hashrate.",
```

Comment (meowsbits) [reply openrpc.json]: yea

**Comment by meowsbits** [comment]
@BelfordZ Can you cite a source documenting the deprecation of this method? I can't seem to find one. 

I have looked here
- https://github.com/ethereum/go-ethereum/search?q=submitHashrate&type=Code
- https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_submithashrate
- https://ethereum.stackexchange.com/questions/2298/failed-to-submit-hashrate-error/2302

**Comment by BelfordZ** [comment]
@meowsbits the topic came up once when working on mining pool stuff. The easiest reason for deprecating is that there is no good reason for using it to begin with.

Consider what 'submitting your hashrate' means.... 

It's useless for pools, they cannot calculate payout based on this, and must do hashrate estimation based on submitted shares and target difficulty. 

It's useless for miners that use pools since the hashrate can be viewed on the pool. If you dont trust the pool, then why would you trust it to store the values you report via 'submitHashrate'?

IMO it seems out of place, and any hashrate reporting should be handled outside of geth if a user wants to track hashrate irrespective of # blocks mined / difficulty.

**Comment by meowsbits** [comment]
SGTM

File: openrpc.json [hunk 2]
```diff
@@ -965,7 +965,8 @@
       },
       {
         "name": "eth_submitHashrate",
-        "summary": "Returns an array of all logs matching a given filter object.",
+        "deprecated": true,
+         "summary": "Used for submitting mining hashrate.",
```
Comment (meowsbits) [inline openrpc.json]:
```suggestion
        "summary": "Used for submitting mining hashrate.",
```

**Pull request merged**